includes/core/class-wc-payments-order.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Class file for WCPay\Core\WC_Payments_Order
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core;
+
+defined( 'ABSPATH' ) || exit;
+
+use WC_Order;
+use WC_Payments_Utils;
+
+/**
+ * WC_Payments_Order
+ */
+class WC_Payments_Order {
+
+	/**
+	 * The WC_Order corresponding to this WC_Payments_Order.
+	 *
+	 * @var WC_Order
+	 */
+	private $order;
+
+	/**
+	 * Constructor when order_id is provided.
+	 *
+	 * @param int $order_id The order Id.
+	 */
+	public function __construct( int $order_id ) {
+		$this->order = wc_get_order( $order_id );
+	}
+
+	/**
+	 * Get the payment metadata for key _intent_id.
+	 *
+	 * @return string
+	 */
+	public function get_intent_id() : string {
+		return $this->order->get_meta( '_intent_id', true );
+	}
+
+	/**
+	 * Set the payment metadata for key _intent_id.
+	 *
+	 * @param string $intent_id The value to be set.
+	 */
+	public function set_intent_id( $intent_id ) {
+		$this->order->update_meta_data( '_intent_id', $intent_id );
+		$this->order->save_meta_data();
+		/**
+		 * Hook: When the order meta data _intent_id is updated.
+		 *
+		 * @since 5.4.0
+		 */
+		do_action( 'wcpay_order_intent_id_updated' );
+	}
+
+	/**
+	 * Get the payment metadata for key _payment_method_id.
+	 *
+	 * @return string
+	 */
+	public function get_payment_method_id() : string {
+		return $this->order->get_meta( '_payment_method_id', true );
+	}
+
+	/**
+	 * Set the payment metadata for key _payment_method_id.
+	 *
+	 * @param  string $payment_method_id The value to be set.
+	 */
+	public function set_payment_method_id( $payment_method_id ) {
+		$this->order->update_meta_data( '_payment_method_id', $payment_method_id );
+		$this->order->save_meta_data();
```

Comment:
Let's see where this will be used, and if the order is saved in that context. This might uncover a bigger issue, which would be that the order object here gets modified, but the original order does not, or vice versa. Is that a problem?